pkg/controllers/state/cluster.go

```diff
@@ -512,3 +513,10 @@ func (c *Cluster) Synchronized(ctx context.Context) error {
 func (c *Cluster) recordConsolidationChange() {
 	atomic.StoreInt64(&c.consolidationState, c.clock.Now().UnixMilli())
 }
+
+func (c *Cluster) LivenessProbe(req *http.Request) error {
+	c.mu.RLock()
```

Comment:
why not take a write lock here?

Reply:
Yeah, that's better. I did the read locks at first to be less intrusive, but I agree that write locks is better at finding the problem quickly.